app/javascript/packs/Components/Filters.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import SelectFilter from './SelectFilter'
+import style from './Filters.module.scss'
+
+class Filters extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleNameChange = this.handleNameChange.bind(this);
+        this.handleRatingChange = this.handleRatingChange.bind(this)
+        this.handleCuisineChange = this.handleCuisineChange.bind(this);
+        this.handleSpeedChange = this.handleSpeedChange.bind(this);
+        this.state = {
+            name: '',
+            rating: 0,
+            cuisine: '',
+            speed:Number.MAX_SAFE_INTEGER
+        };
+    }
+
+    handelChange(o){
+        this.setState((state) => {
+            state[o.key]  = o.value;
+            this.props.onFilterChange(state);
+        });
```

Comment:
Couple of comments here:
1. Try to avoid changing the `state` object itself, define a new object and `return` it in the end.
   Example:
   ```jsx
   this.setState(state => {
     return {
       [o.key]: o.value,
     };
   });
   ```
1. Don't call other functions that have side-effects from within `setState`, that could possibly break something. `setState` provides a callback argument that's called when it finishes updating its state:
   ```jsx
   this.setState(..., () => this.props.onFilterChange(this.state));
  ```